x/arkeo/keeper/providers.go

```diff
@@ -11,7 +11,7 @@ import (
 func (k KVStore) setProvider(ctx cosmos.Context, key string, record types.Provider) {
 	store := ctx.KVStore(k.storeKey)
 	buf := k.cdc.MustMarshal(&record)
-	if buf == nil {
+	if buf == nil || record.Bond.IsZero() {
```

Comment:
I believe this is problematic, and maybe we can add a test to prove / disprove.

If a provider un-bond's but still has active contracts, we still need the record so users can cancel their contracts.